test/cppsuite/src/component/statistics_monitor.cpp

```diff
@@ -0,0 +1,181 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "statistics_monitor.h"
+
+#include <fstream>
+
+#include "perf_plotter.h"
+#include "src/common/constants.h"
+#include "src/common/logger.h"
+#include "src/storage/connection_manager.h"
+#include "statistics/cache_limit.h"
+#include "statistics/database_size.h"
+#include "statistics/statistics.h"
+
+extern "C" {
+#include "test_util.h"
+}
+
+namespace test_harness {
+
+/* Inline methods implementation. */
+
+/*
+ * The WiredTiger configuration API doesn't accept string statistic names when retrieving statistic
+ * values. This function provides the required mapping to statistic id. We should consider
+ * generating it programmatically in `stat.py` to avoid having to manually add a condition every
+ * time we want to observe a new postrun statistic.
+ */
+inline int
+get_stat_field(const std::string &name)
+{
+    if (name == CACHE_HS_INSERT)
+        return (WT_STAT_CONN_CACHE_HS_INSERT);
+    else if (name == CC_PAGES_REMOVED)
+        return (WT_STAT_CONN_CC_PAGES_REMOVED);
+    testutil_die(EINVAL, "get_stat_field: Stat \"%s\" is unrecognized", name.c_str());
+}
+
+/* statistics_monitor class implementation */
```

Comment:
Remove this comment. I think it's was used to separate class definitions when we had many classes in the same file.